### PR TITLE
Fix Resumable upload race conditions

### PR DESCRIPTION
**Pull request by hotzenklotz**
This PR adds unit tests for `Resumable` upload race conditions. It provokes a slow backend response to the first chunk that is being uploaded such that the second "iteration" of Resumbles state logic will trigger the same request again causing a race condition. 

This test fails on the current state of Resumble on `master` - recreating the reace condition/oberserved issue in WK. With the proposed fix in #9509, the tests resolve.

### Steps to test:
- Run unit tests

### Issues:
- related to #9509

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Added changelog entry (create a `$PR_NUMBER.md` file in `unreleased_changes` or use `./tools/create-changelog-entry.py`)
- [ ] Added migration guide entry if applicable (edit the same file as for the changelog)
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: frontend/javascripts/test/libs/resumable-upload-use-cases.spec.ts [hunk 1]
```diff
@@ -610,4 +610,112 @@ describe("Resumable Use Cases (WebKnossos Patterns)", () => {
...
+      // is already in-flight here (response delay=40ms, so it has not finished yet).
+      const inflightAtRaceMoment = backendMock.getInflightUploads();
+
+      releaseChunk1Body();
+      await resumable.waitForComplete();
+      await backendMock.waitForIdle();
+
+      // If race fires: inflightAtRaceMoment=1 (duplicate POST still running).
+      // If fixed:      inflightAtRaceMoment=0 (no spurious upload started).
+      expect(inflightAtRaceMoment).toBe(0);
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply frontend/javascripts/test/libs/resumable-upload-use-cases.spec.ts]: 😢

[the remark this answers is not in the document]
Comment (hotzenklotz) [reply frontend/javascripts/test/libs/resumable-upload-use-cases.spec.ts]: adding `await Promise.resolve();` did nothing. the assertion `expect(inflightAtRaceMoment).toBe(0);` still passes

**Review by philippotto — DISMISSED** [review]
lgtm 👍 whats the next step? merging the fix into this PR?

**Comment by hotzenklotz** [comment]
> lgtm 👍 whats the next step? merging the fix into this PR?

yes

**Comment by philippotto** [comment]
> yes

ok, great, please adapt pr title and changelog. should I test the PR again on a dev instance or did you do that?

**Comment by hotzenklotz** [comment]
> should I test the PR again on a dev instance or did you do that?

Please test again.

**Review by philippotto — APPROVED** [review]
uploading worked for me (I slowed down the network traffic, turned it offline to have some requests failed, turned online again and everything succeeded eventually).
